### PR TITLE
Security

### PR DESCRIPTION
**Pull request by ScifiAN**

### CODE REVIEW

File: Project-security/csrf/my-site/routes/demo.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+});
+
+router.post('/signup', async function (req, res) {
+  const userData = req.body;
+  const enteredEmail = userData.email; // userData['email']
+  const enteredConfirmEmail = userData['confirm-email'];
+  const enteredPassword = userData.password;
+
+  if (
+    !enteredEmail ||
```
Comment (Kameq) [inline Project-security/csrf/my-site/routes/demo.js:51]: again multiline conditions ...
remember to alway use `||` etc at the begining of line (from second line ofcourse)

Comment (Kameq) [reply Project-security/csrf/my-site/routes/demo.js:51]: we will need to think about `lint` later on in this case

File: Project-security/db-injection/routes/discussion.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+router.get('/discussion', async function (req, res) {
+  let filter = '';
+
+  if (req.query.author) {
+    filter = `WHERE author = ?`; 
+  }
+
+  const query = `SELECT * FROM comments ${filter}`;
+
+  console.log(query);
```
Comment (Kameq) [inline Project-security/db-injection/routes/discussion.js]: console.log - remove

File: Project-security/xss/app.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+app.set('view engine', 'ejs');
+app.set('views', path.join(__dirname, 'views'));
+
+app.use(express.static('public'));
+app.use(express.urlencoded({ extended: false }));
+
+app.use(discussionRoutes);
+
+app.use(function(error, req, res, next) {
+  console.log(error);
```
Comment (Kameq) [inline Project-security/xss/app.js]: console.log

**Review by Kameq — CHANGES_REQUESTED** [review]
I have a feeling that in this Pull request there are several different projects (perhaps from previous pull requests)

anyway its best to do Pull Request for single project as it is very confusing right now

File: Project-security/csrf/my-site/routes/demo.js (hunk 1 above)
Comment (ScifiAN) [reply Project-security/csrf/my-site/routes/demo.js:51]: Well it was project included in a course that show what this kind of security problems there are. So basically it was 3 mini projects in 1 large project.

**Pull request merged**